app/main.py

```diff
@@ -2,0 +1,47 @@
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None :
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None :
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        for skill in ["JavaScript", "HTML", "CSS"]:
+            self.learn_skill(skill)
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None :
+        super().__init__(name)
+        for skill in ["Python", "SQL", "Django"]:
+            self.learn_skill(skill)
```

Comment:
Consider using `self.skills.extend(['Python', 'SQL', 'Django'])` instead of the loop to initialize skills for better performance and readability.